UnityEngine.AssetBundles/Editor/Tests/ABModelTests.cs

```diff
@@ -45,25 +45,43 @@ public void TearDown()
     [Test]
     public void AddBundlesToUpdate_AddsCorrectBundles_ToUpdateQueue()
     {
+        // Account for existing asset bundles
+        int numBundles = ABModelUtil.BundlesToUpdate.Count;
+
         Model.AddBundlesToUpdate(m_BundleInfo);
-        Assert.AreEqual(3, ABModelUtil.BundlesToUpdate.Count);
+        Assert.AreEqual(numBundles + 3, ABModelUtil.BundlesToUpdate.Count);
     }
 
     [Test]
     public void Update_ReturnsTrue_ForRepaintOnFinalElement()
```

Comment:
this test is no longer functioning (though it did have pre-existing issues). For one, inside your for loop you have ``` if(i < numChildren) ``` which is always true because if it isn't you'll leave the for loop. For two, you're looping based on numChildren which is unrelated to the test. As it is was, this test would only pass in a project with no other bundles. As it is now, it will only pass if you're in a project with three or fewer bundles.  Both are obviously broken. I haven't totally dug into it, but I think before you do the add bundles, you need to:
```
        for (int i = 0; i <= numChildren; ++i)
        {
            if (Model.Update())
                break;
        }
```
to clear out whatever data is there.  Then do the add to update, then just put back the same code that was there before (two falses and a true). But i haven't tested this idea with multiple setups yet.